components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/loader/UIBasedConfigurationLoader.java

```diff
@@ -300,14 +304,12 @@ private void loadStepAuthenticator(StepConfig stepConfig, IdentityProvider idp,
             authenticatorConfig = new AuthenticatorConfig();
             authenticatorConfig.setName(authenticatorName);
 
-            for (ApplicationAuthenticator appAuthenticator : FrameworkServiceComponent.getAuthenticators()) {
-
-                if (authenticatorName.equalsIgnoreCase(appAuthenticator.getName())) {
-                    authenticatorConfig.setApplicationAuthenticator(appAuthenticator);
-                    break;
-                }
+            ApplicationAuthenticator appAuthenticatorForConfig = ApplicationAuthenticatorManager.getInstance()
+                    .getApplicationAuthenticatorByName(authenticatorName, tenantDomain);
```

Comment:
The original logic ignore application authenticator name case. Is this handled ?